tests/test_non_persistent_deployment.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+import pytest
+import os
+import torch
+from types import SimpleNamespace
+from .utils import *  # noqa: F401
+import mii
+''' These fixtures provide a local deployment and ensure teardown '''
+
+
+@pytest.fixture(scope="function")
+def mii_configs(
+    tmpdir: str,
+    dtype: str,
+    tensor_parallel: int,
+    load_with_sys_mem: bool,
+    enable_load_balancing: bool,
+):
+
+    # Create a hostfile for DeepSpeed launcher when load_balancing is enabled
+    hostfile = os.path.join(tmpdir, "hostfile")
+    num_gpu = torch.cuda.device_count()
+    enable_load_balancing = enable_load_balancing
+    if enable_load_balancing:
+        with open(hostfile, "w") as f:
+            f.write(f"localhost slots={num_gpu}")
+
+    return {
+        'dtype': dtype,
+        'tensor_parallel': tensor_parallel,
```

Comment:
Can we run with tensor parallelism with this deployment type? I think we will have to run with `deepspeed --num_gpus X mii_example.py` for tensor parallelism to work with this new deployment type. We should add a check in `deployment.py` that asserts that `int(os.genenv["WORLD_SIZE"]) == mii_config.tensor_parallel` for the non-persistent deployment only.